api/test/unit/utils/test_image_proxy.py

```diff
@@ -446,13 +499,19 @@ def test_photon_get_raises_by_not_allowed_types(image_type):
         (None, b"unknown"),
     ],
 )
-def test_photon_get_saves_image_type_to_cache(redis, headers, expected_cache_val):
-    image_url = TEST_IMAGE_URL.replace(".jpg", "")
-    image = ImageFactory.create(url=image_url)
+async def test_photon_get_saves_image_type_to_cache(
+    pook_off, image_url, redis, headers, expected_cache_val
+):
+    image_url = image_url.replace(".jpg", "")
+    image = await sync_to_async(ImageFactory.create)(url=image_url)
+
+    pook.on()
+    mock_head = pook.head(image_url).reply(200)
+    if headers:
+        mock_head = mock_head.headers(headers)
 
-    with mock.patch("requests.head") as mock_head, pytest.raises(UnsupportedMediaType):
-        mock_head.return_value.headers = headers
-        photon_get(image_url, image.identifier)
+    with pytest.raises(UnsupportedMediaType):
+        await photon_get(image_url, image.identifier)
```

Comment:
Using pook instead of mocking aiohttp is a more reliable way to prevent the head request from going through.